internal/cli/users_roles.go

```diff
@@ -63,3 +80,119 @@ func showUserRolesCmd(cli *cli) *cobra.Command {
 
 	return cmd
 }
+
+func addUserRolesCmd(cli *cli) *cobra.Command {
+	var inputs userRolesInput
+
+	cmd := &cobra.Command{
+		Use:     "assign",
+		Aliases: []string{"add", "associate"},
+		Args:    cobra.MaximumNArgs(1),
+		Short:   "Assign roles to a user",
+		Long:    "Assign existing roles to a user.",
+		Example: `  auth0 users roles assign <user-id>
+  auth0 users roles associate <user-id> --roles <role-id1,role-id2>
+  auth0 users roles add <user-id> -r "rol_1eKJp3jV04SiU04h,rol_2eKJp3jV04SiU04h"`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				if err := userID.Ask(cmd, &inputs.ID); err != nil {
+					return err
+				}
+			} else {
+				inputs.ID = args[0]
+			}
+
+			if len(inputs.Roles) == 0 {
+				if err := cli.pickUserRoles(&inputs); err != nil {
+					return err
+				}
+			}
+
+			var rolesToAssign []*management.Role
+			for _, roleID := range inputs.Roles {
+				rolesToAssign = append(rolesToAssign, &management.Role{
+					ID: auth0.String(roleID),
+				})
+			}
+
+			if err := ansi.Waiting(func() (err error) {
+				return cli.api.User.AssignRoles(inputs.ID, rolesToAssign)
+			}); err != nil {
+				return fmt.Errorf("failed to assign roles for user with ID %s: %w", inputs.ID, err)
+			}
+
+			var userRoleList *management.RoleList
+			if err := ansi.Waiting(func() (err error) {
+				userRoleList, err = cli.api.User.Roles(inputs.ID)
+				return err
+			}); err != nil {
+				return fmt.Errorf("failed to find roles for user with ID %s: %w", inputs.ID, err)
+			}
+
+			cli.renderer.UserRoleList(userRoleList.Roles)
+
+			return nil
+		},
+	}
+
+	userRoles.RegisterStringSlice(cmd, &inputs.Roles, nil)
+	cmd.Flags().BoolVar(&cli.json, "json", false, "Output in json format.")
+
+	return cmd
+}
+
+func (cli *cli) pickUserRoles(inputs *userRolesInput) error {
+	var currentUserRoleList *management.RoleList
+	if err := ansi.Waiting(func() (err error) {
+		currentUserRoleList, err = cli.api.User.Roles(inputs.ID)
+		return err
+	}); err != nil {
+		return fmt.Errorf("failed to find the current roles for user with ID %s: %w", inputs.ID, err)
+	}
+
+	var roleList *management.RoleList
+	if err := ansi.Waiting(func() (err error) {
+		roleList, err = cli.api.Role.List()
+		return err
+	}); err != nil {
+		return fmt.Errorf("failed to list all roles: %w", err)
+	}
+
+	if len(roleList.Roles) == len(currentUserRoleList.Roles) {
+		return fmt.Errorf("the user with ID %q already has all possible roles assigned", inputs.ID)
+	}
+
+	const emptySpace = " "
+	var options []string
+	for _, role := range roleList.Roles {
+		if !containsRole(currentUserRoleList.Roles, role.GetID()) {
+			options = append(options, fmt.Sprintf("%s%s(Name: %s)", role.GetID(), emptySpace, role.GetName()))
+		}
+	}
+
+	rolesPrompt := &survey.MultiSelect{
```

Comment:
Consider moving this to https://github.com/auth0/auth0-cli/blob/main/internal/cli/input.go, now that it's going to be used in 3 commands.